include/clap/ext/draft/location.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "../../plugin.h"
+#include "../../color.h"
+
+// This extension allows a host to tell the plugin more about its position
+// within a project or session.
+
+static CLAP_CONSTEXPR const char CLAP_EXT_LOCATION[] = "clap.location/1";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+   // Represents a document/project/session.
+   CLAP_PLUGIN_LOCATION_PROJECT,
+
+   // Represents a group of tracks.
+   // It can contain both track groups, tracks and devices (post processing).
+   // The first device within a track group has the index of
+   // the last track or track group within this group + 1.
+   CLAP_PLUGIN_LOCATION_TRACK_GROUP,
+
+   // Represents a single tracks. It contains devices (serial).
+   CLAP_PLUGIN_LOCATION_TRACK,
+
+   // Represents a single device.
+   // It contains other nested device chains.
+   CLAP_PLUGIN_LOCATION_DEVICE,
+
+   // Represents a nested device chain (serial).
+   // Its parent must be a device.
+   // It contains other devices.
+   CLAP_PLUGIN_LOCATION_NESTED_DEVICE_CHAIN,
+};
+
+typedef struct clap_plugin_location_element {
+   // Kind of the element, must be one of the CLAP_PLUGIN_LOCATION_* values.
+   uint32_t kind;
+
+   // Index within the parent element.
+   // Set to 0 if irrelevant.
+   uint32_t index;
+
+   // Internal ID of the element.
+   // This is not intended for display to the user,
+   // but rather to give the host a potential quick way for lookups.
+   const char *id;
+
+   // User friendly name of the element.
+   const char *name;
```

Comment:
what's the lifetime of these const char *? Any reason we didn't use `char id[CLAP_TEXT_SIZE]` like we do most other places?

Reply:
Switching to char buffers, I think it makes it simpler you're right.